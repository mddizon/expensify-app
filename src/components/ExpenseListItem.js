import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ id, dispatch, description, amount, createdAt }) => (
  <div>
    <h3>Expense: {description}</h3>
    <p>Amount: {amount}</p>
    <p>CreatedAt: {createdAt}</p>
    <Link to={`/edit/${id}`}>Edit</Link>
  </div>
);

export default ExpenseListItem;
