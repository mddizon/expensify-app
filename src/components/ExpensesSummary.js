import React from 'react';
import { connect } from 'react-redux';
import numeral from "numeral";
import selectExpenses from '../selectors/expenses';
import getExpenseTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({num_expenses, total_expenses}) => {
  const word = num_expenses === 1 ? 'expense' : 'expenses';
  return (
  <div>
    <h3>
      Viewing {num_expenses} {word} totalling {numeral(total_expenses/100).format('$0,0.00')}
    </h3>
  </div>
)};

const mapStateToProps = (state) => {
  const expenses = selectExpenses(state.expenses, state.filters)
  return {
    num_expenses: expenses.length || 0,
    total_expenses: getExpenseTotal(expenses) || 0
  };
};

export default connect(mapStateToProps)(ExpensesSummary);

