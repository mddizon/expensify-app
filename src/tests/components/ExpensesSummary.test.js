import React from 'react';
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test('should render ExpensesSummary correctly with no expenses', () => {
  const wrapper = shallow(<ExpensesSummary />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with one expense', () => {
  const wrapper = shallow(<ExpensesSummary num_expenses="1" total_expenses="123"/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary num_expenses="2" total_expenses="123"/>);
  expect(wrapper).toMatchSnapshot();
});

