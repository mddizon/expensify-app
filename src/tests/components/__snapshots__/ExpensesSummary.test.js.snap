// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h3>
    Viewing 
    2
     
    expenses
     totalling 
    $1.23
  </h3>
</div>
`;

exports[`should render ExpensesSummary correctly with no expenses 1`] = `
<div>
  <h3>
    Viewing 
     
    expenses
     totalling 
    $0.00
  </h3>
</div>
`;

exports[`should render ExpensesSummary correctly with one expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expenses
     totalling 
    $1.23
  </h3>
</div>
`;
